fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneIndexFailureTest.java

```diff
@@ -273,106 +272,122 @@ void repartitionAndMerge(boolean useLegacyAsyncToSync) throws IOException {
                 segmentCounts);
 
         try (FDBRecordContext context = openContext(contextProps)) {
```

Comment:
You could probably just call:
```java
dataModel.validate(() -> openContext(contextProps));
```
It will assert that the documents align, and that the partitions are all an acceptable size. It is less strict than the assertions here, but it is probably less brittle in the long run.

Reply:
Do you mean to replace the entire `try` block with both `validateDocsInPartition` in it? In that case, it would probably also make sense to change the original in `LuceneIndexTest`?